File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String[] msg=logLine.split(":");
        return msg[1].trim();
    }

    public static String logLevel(String logLine) {
        String[] msg=logLine.split(":");
        return msg[0].substring(1,msg[0].length()-1).toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine).concat(" (".concat(logLevel(logLine).concat(")")));
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string log){
        std::size_t found = log.find_first_of(": ");
        std::string linelog=log.substr(found+2,log.length()-(found+2));
        return linelog;
    }
    std::string log_level(std::string log){
        std::size_t found_start = log.find_first_of("[");
        std::size_t found_end = log.find_first_of("]");
        std::string linelog=log.substr(found_start+1,found_end-found_start-1);
        return linelog;
    }
    std::string reformat(std::string log){
        std::string linelog=log_line::message(log);
        linelog.append(" (");
        linelog.append(log_line::log_level(log));
        linelog.append(")");
        return linelog;
    }
    

} // namespace log_line
